core/include/scattered_array.h

```diff
@@ -84,7 +90,7 @@
  */
 #if defined(X86_64)
```

Comment:
Do we need to remove #if defined(X86_64) here?

Reply:
Done, removed.